File: EduHubConnect-Backend-UsingSpringBoot/src/main/java/com/connect/eduHub/model/Comment.java
package com.connect.eduHub.model;

import java.time.LocalDateTime;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Comment {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String content;

    @ManyToOne
    private Post post;

    @ManyToOne
    private User user;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: EduHubConnect-AndroidFrontEnd/app/src/main/java/com/connect/eduhubconnect/model/Comment.kt
package com.connect.eduhubconnect.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "comments")
data class Comment(
    @PrimaryKey val id: Long,
    val content: String,
    val userId: Long,
    val postId: Long,
    val userName: String,
    val userProfileImage:String,
    val userJobTitle:String,
    val createdAt: String,
    val updatedAt:String
)
